### PR TITLE
feat: Adding bcrypt support for argocd CLI

### LINKED ISSUE #10050: Expose `bcrypt` as part of `argocd`
# Summary

If `bcrypt` is an important component of how `argocd` works, and argo-cd (the project) doesn't believe everyone has easy access to `bcrypt` (I'm willing to believe this, although at this point, it might be possible to argue that everyone has `python` or even access to `docker`), then `argocd` (the program) should offer `bcrypt` as a command that can be run to do whatever is needed to perform a `bcrypt`.

# Motivation

https://argo-cd.readthedocs.io/en/stable/faq/#i-forgot-the-admin-password-how-do-i-reset-it

> To change the password, edit the argocd-secret secret and update the admin.password field with a new bcrypt hash. You can use a site like https://www.browserling.com/tools/bcrypt to generate a new hash.

One should *never* suggest a website for something relating to credentials. It's always the wrong answer.

# Proposal

Add a `argocd bcrypt` that supports performing whatever action this series of steps imagines a user needs to do.

---

Disclaimer: At this point, I am not remotely certain these steps are meaningful as they certainly don't correspond to the structure of my instances.

### PR DESCRIPTION
**Pull request by ameytotawar**
Fixes #10050

This PR adds the bcrypt support for argocd CLI under `argocd admin` command to provide capability to bcrypt admin password. Earlier there was a 3rd party recommendation for the same.

#Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
This is awesome! Can you update the bcrypt instructions here to ask people to use the new `argocd admin bcrypt` command? https://argo-cd.readthedocs.io/en/latest/faq/#i-forgot-the-admin-password-how-do-i-reset-it

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Could we nest this command under `account` instead of `admin`? Sorry, I think I misspoke when we discussed that.

**Comment by ameytotawar** [comment]
> This is awesome! Can you update the bcrypt instructions here to ask people to use the new `argocd admin bcrypt` command? https://argo-cd.readthedocs.io/en/latest/faq/#i-forgot-the-admin-password-how-do-i-reset-it

* Thanks. I have updated the faq docs

> Could we nest this command under `account` instead of `admin`? Sorry, I think I misspoke when we discussed that.

* Nested the bcrypt cmd under account instead of admin.

File: docs/faq.md [hunk 1]
```diff
@@ -44,11 +44,9 @@ a secret named `argocd-initial-admin-secret`.
 
 To change the password, edit the `argocd-secret` secret and update the `admin.password` field with a new bcrypt hash.
 
-!!! note "Generating a bcrypt hash"
-    Use a trustworthy, offline `bcrypt` implementation such as the [Python bcrypt library](https://pypi.org/project/bcrypt/) to generate the hash.
+  Use the following command to generate a bcrypt hash for `admin.password`
```
Comment (crenshaw-dev) [inline docs/faq.md:47]:
```suggestion
!!! note "Generating a bcrypt hash"
    Use the following command to generate a bcrypt hash for `admin.password`
```

I suggest we leave this in a nice, loud box. You can preview it here: https://argo-cd--10934.org.readthedocs.build/en/10934/faq/#i-forgot-the-admin-password-how-do-i-reset-it

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by crenshaw-dev — APPROVED** [review]
lgtm, thanks @ameytotawar!

**Pull request merged**